lib/libnfs.c

```diff
@@ -2238,3 +2383,59 @@ rpc_null_task(struct rpc_context *rpc, int program, int version, rpc_cb cb,
 
 	return pdu;
 }
+
+#ifdef HAVE_TLS
+/*
+ * Call this in place of rpc_null_task() if user wants TLS security for the RPC
+ * connection. Since we only ever send AUTH_TLS NULL RPC for NFS_PROGRAM it does
+ * not need the program parameter.
+ * Callback 'cb' is called not after we get the reply for this AUTH_TLS NULL RPC,
+ * but after the TLS handshake completes (success or failure).
+ */
+struct rpc_pdu *
```

Comment:
ONC-RPC over TLS can in theory be used also for other protocols than nfs 4.x, yes?

In that case, would it make sense to add a 'int program' argument here
just so that if someone in the future that wants to add TLS support for other protocols?
Or is that NFSv4.x is unique in the way that it does not rely on portmapper the reason?
I ask from ignorance since I don't know how/if running PORTMAPPER would even be possible across TLS.

Not asking for a change, only to educate myself.

Reply:
Technically since it secures the RPC transport it can be used by any RPC program. Since Linux NFS server is the first implementation of RPC-with-TLS I'm taking the liberty of aligning with its behaviour. Linux NFS server today doesn't use secure RPC channel for anything other than the NFS program, i.e., neither PORTMAP nor MOUNT programs use it, so I'd say let's go with this initially, since at least that makes us interoperable with the only server implementation we can get our hands on. 
Do you take part in any connectathon or any other interop events? Maybe you or someone can then try interop'ing libnfs with other NFS servers that support RPC-with-TLS and based on that input we can make it configurable. For now I'd leave it like this.